document/macros.rst

```diff
@@ -0,0 +1,171 @@
+.. |dataseg| mathmacro:: \href{../syntax/modules.html#syntax-dataseg}{\mathit{dataseg}}
+
+.. |sec| mathmacro:: \mathit{sec}
+.. |typesec| mathmacro:: \mathit{typesec}
+.. |importsec| mathmacro:: \mathit{importsec}
+.. |funcsec| mathmacro:: \mathit{funcsec}
+.. |tablesec| mathmacro:: \mathit{tablesec}
+.. |memsec| mathmacro:: \mathit{memsec}
+.. |globalsec| mathmacro:: \mathit{globalsec}
+.. |exportsec| mathmacro:: \mathit{exportsec}
+.. |startsec| mathmacro:: \mathit{startsec}
+.. |elemsec| mathmacro:: \mathit{elemsec}
+.. |codesec| mathmacro:: \mathit{codesec}
+.. |datasec| mathmacro:: \mathit{datasec}
+.. |customsec| mathmacro:: \mathit{customsec}
+
+.. |locals| mathmacro:: \mathit{locals}
+.. |code| mathmacro:: \mathit{code}
+
+.. |typeidx| mathmacro:: \href{../syntax/modules.html#syntax-typeidx}{\mathit{typeidx}}
+.. |funcidx| mathmacro:: \href{../syntax/modules.html#syntax-funcidx}{\mathit{funcidx}}
+.. |tableidx| mathmacro:: \href{../syntax/modules.html#syntax-tableidx}{\mathit{tableidx}}
+.. |memidx| mathmacro:: \href{../syntax/modules.html#syntax-memidx}{\mathit{memidx}}
+.. |globalidx| mathmacro:: \href{../syntax/modules.html#syntax-globalidx}{\mathit{globalidx}}
+.. |localidx| mathmacro:: \href{../syntax/modules.html#syntax-localidx}{\mathit{localidx}}
+.. |labelidx| mathmacro:: \href{../syntax/modules.html#syntax-labelidx}{\mathit{labelidx}}
+
+.. |MODULE| mathmacro:: \mathsf{module}
+.. |TYPE| mathmacro:: \mathsf{type}
+.. |FUNC| mathmacro:: \mathsf{func}
+.. |TABLE| mathmacro:: \mathsf{table}
+.. |MEM| mathmacro:: \mathsf{mem}
+.. |GLOBAL| mathmacro:: \mathsf{global}
+.. |LOCAL| mathmacro:: \mathsf{local}
+.. |LABEL| mathmacro:: \mathsf{label}
+.. |IMPORT| mathmacro:: \mathsf{import}
+.. |EXPORT| mathmacro:: \mathsf{export}
+.. |CODE| mathmacro:: \mathsf{code}
+.. |DATA| mathmacro:: \mathsf{data}
+.. |ELEM| mathmacro:: \mathsf{elem}
+.. |START| mathmacro:: \mathsf{start}
+.. |CUSTOM| mathmacro:: \mathsf{custom}
+
+.. |INDEX| mathmacro:: \mathsf{index}
+.. |VALUE| mathmacro:: \mathsf{value}
+.. |INIT| mathmacro:: \mathsf{init}
+.. |BODY| mathmacro:: \mathsf{body}
+.. |NAME| mathmacro:: \mathsf{name}
+.. |DESC| mathmacro:: \mathsf{desc}
+.. |OFFSET| mathmacro:: \mathsf{offset}
+.. |ALIGN| mathmacro:: \mathsf{align}
+
+.. |TYPES| mathmacro:: \mathsf{types}
+.. |FUNCS| mathmacro:: \mathsf{funcs}
+.. |TABLES| mathmacro:: \mathsf{tables}
+.. |MEMS| mathmacro:: \mathsf{mems}
+.. |GLOBALS| mathmacro:: \mathsf{globals}
+.. |LOCALS| mathmacro:: \mathsf{locals}
+.. |LABELS| mathmacro:: \mathsf{labels}
+.. |IMPORTS| mathmacro:: \mathsf{imports}
+.. |EXPORTS| mathmacro:: \mathsf{exports}
+
+.. |UNREACHABLE| mathmacro:: \mathsf{unreachable}
+.. |NOP| mathmacro:: \mathsf{nop}
+.. |BLOCK| mathmacro:: \mathsf{block}
+.. |LOOP| mathmacro:: \mathsf{loop}
+.. |IF| mathmacro:: \mathsf{if}
+.. |ELSE| mathmacro:: \mathsf{else}
+.. |END| mathmacro:: \mathsf{end}
+.. |BR| mathmacro:: \mathsf{br}
+.. |BRIF| mathmacro:: \mathsf{br\_if}
```

Comment:
And in any case, preserving the underscores in the instruction names that have them would improve readability of the source, and searchability as well.